imutils/video/webcamvideostream.py

```diff
@@ -3,10 +3,15 @@
 import cv2
 
 class WebcamVideoStream:
-	def __init__(self, src=0, name="WebcamVideoStream"):
+	def __init__(self, src=0, cap=None, resolution=(320, 240),
+		name="WebcamVideoStream"):
 		# initialize the video camera stream and read the first frame
 		# from the stream
-		self.stream = cv2.VideoCapture(src)
+		self.stream = cap
+		if cap is not None:
```

Comment:
If you're going to have the `cap` variable, shouldn't this read `if cap is None`?

Reply:
Yes, good catch. If the `cap` is passed into this method, then we don't want assign the stream or set any properties. Only `if cap is None` do we want to assign the stream and set the properties on it.